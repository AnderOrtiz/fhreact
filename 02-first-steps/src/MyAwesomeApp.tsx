import type { CSSProperties } from "react"

const firtName = 'Ander',
    lastName = 'Ortiz',
    favoriteGames = ['Elden Ring', 'Smash', 'Shovel knight'],
    isActive = true,
    address = {
        zipCode: 'ABC',
        country: 'Canad√°',
    },
    myStyle:CSSProperties = {
                backgroundColor: '#000',
                color:'white',
                borderRadius: isActive? 10:2,
                padding:10,
                marginTop:30,
            }

export function MyAwasomeApp() {
    return (
        <>
            <h1 data-testid="first-name-title" >{firtName}</h1>
            <h3>{lastName}</h3>

            <p>{favoriteGames.join(', ')}</p>
            <p>{2 + 2}</p>

            <h1>{isActive ? 'Activo' : 'No activo'}</h1>
            <p style={myStyle}>{JSON.stringify(address)}</p>
        </>
    )
}