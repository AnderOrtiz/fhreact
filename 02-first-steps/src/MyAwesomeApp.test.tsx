import { describe, expect, test } from "vitest";
import { render, screen } from "@testing-library/react";


import { MyAwasomeApp } from "./MyAwesomeApp";

describe('MyAwesomeApp', () => {
    test('should render first name y lastname', () => {

        const { container } = render(<MyAwasomeApp />);
        // screen.debug()

        const h1 = container.querySelector('h1'),
            h3 = container.querySelector('h3')

        expect(h1?.innerHTML).toContain('Ander')
        expect(h3?.innerHTML).toContain('Ortiz')
    });


    test('should render first name y lastname', () => {
        render(<MyAwasomeApp />)
        screen.debug();

        // const h1 = screen.getByRole('heading', {
        //     level:1
        // });

        const h1 = screen.getByTestId('first-name-title');

        expect(h1.innerHTML).toContain('Ander');

    });


    test('should match snapshot', () => {
        const { container } = render(<MyAwasomeApp />);
        expect(container).toMatchSnapshot();
    })

    test('should match snapshot', () => {
        render(<MyAwasomeApp />);
        expect(screen.getByTestId('div-app')).toMatchSnapshot();
    })
})


/**Se recomineda usar conteiner para manipulaciÃ³nes iniciales sin evento
 * pero es recomendable adaptarse a usar screen
 * 
 */