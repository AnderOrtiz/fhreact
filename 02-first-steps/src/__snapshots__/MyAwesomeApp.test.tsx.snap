// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapshot 1`] = `
<div>
  <div
    data-testid="div-app"
  >
    <h1
      data-testid="first-name-title"
    >
      Ander
    </h1>
    <h3>
      Ortiz
    </h3>
    <p
      class="mi-clase-favorita"
    >
      Elden Ring, Smash, Shovel knight
    </p>
    <p>
      4
    </p>
    <h1>
      Activo
    </h1>
    <p
      style="background-color: rgb(0, 0, 0); color: white; border-radius: 10px; padding: 10px; margin-top: 30px;"
    >
      {"zipCode":"ABC","country":"Canadá"}
    </p>
  </div>
</div>
`;

exports[`MyAwesomeApp > should match snapshot 2`] = `
<div
  data-testid="div-app"
>
  <h1
    data-testid="first-name-title"
  >
    Ander
  </h1>
  <h3>
    Ortiz
  </h3>
  <p
    class="mi-clase-favorita"
  >
    Elden Ring, Smash, Shovel knight
  </p>
  <p>
    4
  </p>
  <h1>
    Activo
  </h1>
  <p
    style="background-color: rgb(0, 0, 0); color: white; border-radius: 10px; padding: 10px; margin-top: 30px;"
  >
    {"zipCode":"ABC","country":"Canadá"}
  </p>
</div>
`;
